docs/rules/custom-rules.md

```diff
@@ -131,61 +162,97 @@ Notice we don't need to include `property` in this approach.
 ```yaml
 rules:
   assert/path-item-operation-required:
-    subject: PathItem
+    subject:
+      type: PathItem
+    assertions:
+      required:
+        - get
     message: Every path item must have a GET operation.
-    required:
-      - get
 ```
 
 The following example asserts that Tags have both name and description defined.
 
 ```yaml
 rules:
   assert/tag-name-and-desc-defined:
-    subject: Tag
-    property:
-      - name
-      - description
+    subject:
+      type: Tag
+      property:
+        - name
+        - description
+    assertions:
+      defined: true
     message: Every tag must have a name and description.
-    defined: true
 ```
 
 Another way to compose that rule is to require the subject keys:
 
 ```yaml
 rules:
   assert/tag-name-and-desc-required:
-    subject: Tag
-    message: Every tag must have a name and description.
-    required:
-      - name
-      - description
+    subject:
+      type: Tag
+    assertions:
+      required:
+        - name
+        - description
 ```
 
-### `context` example
+### `where` example
 
 The following example asserts that PUT responses with HTTP status 200 or 201 cannot return an `application/pdf`content type.
-Without the `context`, the assertion would evaluate every MediaTypesMap including:
-- Responses with all codes, including codes other than 200 or 201
+Without the `where`, the assertion would evaluate every `MediaTypesMap` property including:
+- Responses with all codes, including codes other than 200 or 201.
 - Responses for all HTTP methods, including DELETE, GET, POST, and more.
-To restrict the evaluation, use the `context` feature to limit what will be evaluated.
+To restrict the evaluation, use the `where` feature to limit what is evaluated.
 
 ```yaml
 assert/no-pdf-in-ok-response:
-  context:
-  - type: Operation
-    matchParentKeys: [put]
-  - type: Response
-    matchParentKeys: ['201', '200']
-  subject: MediaTypesMap
-  disallowed: ['application/pdf']
+  where:
+  - subject:
+      type: Operation
+      filterByParentKeys: 
+        - put
+    assertions:
```

Comment:
I would not want to have a `default: true` as a default because it is not in line with default values throughout our configuration file.

Reply:
I see. Anyway, this particular example doesn't work at the moment since the `defined` assertion requires a single `property` to run against. Otherwise it falls into this error: https://github.com/Redocly/redocly-cli/pull/894/files#diff-8ca1f03a8561336a37c8a4f40e18491007a433f3144655ea5cde722aaf32e8f8R75-R79

Reply:
I adjusted the samples. Also, I couldn't understand the logic of what runs on a key?

https://github.com/Redocly/redocly-cli/blob/main/packages/core/src/rules/common/assertions/asserts.ts#L17

For example, what does the `enum` assertion meant when it is run on a key? And what is as an example use case?

Reply:
I see. It would be very hard to find out this fix for a newbie 😄 

When an assertion is supposed to run on keys set it means it could be applied to key names in a Node. Another case is when it could be applied to a particular value (then it requires a `property` to be defined). 

Example:

```yaml

subject: 
  type: PathsMap
assertions:
  enum: 
  - /pets
  - /store
```

which means you can only use the abovementioned keys.

Reply:
FYI I renamed `PathsMap` to `Paths`. #906